datafusion/expr/src/expr.rs

```diff
@@ -3264,9 +3266,21 @@ impl Display for Expr {
             }
             Expr::ScalarVariable(_, var_names) => write!(f, "{}", var_names.join(".")),
             Expr::Literal(v, metadata) => {
-                match metadata.as_ref().map(|m| m.is_empty()).unwrap_or(true) {
-                    false => write!(f, "{v:?} {:?}", metadata.as_ref().unwrap()),
-                    true => write!(f, "{v:?}"),
+                match metadata.as_ref().and_then(|m| m.logical_type()) {
```

Comment:
Pretty printing for Logical Plan. Could be a bit decluttered / imrpoved but does it's job for the sake of this PR.